tests/clients/test_moon_client.py

```diff
@@ -20,12 +20,64 @@ def test_setting_parameters(get_client: MoonClient) -> None:  # noqa
     new_params = [layer_weights + 0.1 for layer_weights in params]
     moon_client.set_parameters(new_params, config)
 
+    # Global model parameters should match those of the new_params
+    global_model_params = [val.cpu().numpy() for _, val in moon_client.global_model.state_dict().items()]
+    # Make sure the MOON model parameters are equal to the global model parameters
+    new_moon_model_params = [val.cpu().numpy() for _, val in moon_client.model.state_dict().items()]
+
+    for i in range(len(params)):
+        assert (new_params[i] == global_model_params[i]).all()
+        assert (new_params[i] == new_moon_model_params[i]).all()
+
```

Comment:
Maybe we can add an assertion here to check whether global_model and old_models_list[-1] is different. (That's what I figured out the bug from)

Reply:
So I can't do it in this part, because I assume that `len(moon_client.old_models_list) == 0` just below. However, below, on both lines 59 and 78, there is an assertion that they are different. I'll add in setting the model twice in the last check, just to be certain.